src/feature/hs/hs_dos.c

```diff
@@ -7,14 +7,84 @@
 
 #define HS_DOS_PRIVATE
 
+#include "core/or/or.h"
+#include "app/config/config.h"
+
 #include "core/or/circuitlist.h"
 
+#include "feature/nodelist/networkstatus.h"
+#include "feature/relay/routermode.h"
+
+#include "lib/evloop/token_bucket.h"
+
 #include "hs_dos.h"
 
+/* Default values. */
+#define HS_DOS_INTRODUCE_CELL_RATE_PER_SEC 25
+#define HS_DOS_INTRODUCE_CELL_BURST_PER_SEC 200
+
+/* Consensus parameters. */
+static uint32_t hs_dos_introduce_rate_per_sec;
+static uint32_t hs_dos_introduce_burst_per_sec;
+
+/* Return the parameter for the introduction rate per sec. */
+static uint32_t
+get_param_rate_per_sec(const networkstatus_t *ns)
+{
+  return networkstatus_get_param(ns, "HSDoSIntroduceRate",
```

Comment:
This seems to be a different consensus parameter than the one defined in prop305. How will these consensus parameter cooperate with the ones from prop305?

Reply:
Yes. Fixup commit: daecc42b37

Reply:
LGTM!